### DISCUSSION TITLE
Weviate +  Gradio UI + Upload Files

### DISCUSSION THREAD

**Discussion by leogomide** [imartinez/privateGPT#581] [Ideas]
Greetings guys, i have seen some videos on youtube about this repository, but im a javacript developer, i have done some learning code in python, but i dont have too much knowledge to build this myself (will take too much time xD)

So, i want to share an idea that i have from 3 videos i have seen recently:

1 - Use weviate free vector database to index the documents

As shown on this video from FireShip (https://www.youtube.com/watch?v=mBcBoGhFndY), weviate has a free vector database that can process both text and image files, the idea is to use it to store the chunks and them query them for the model to use

2 -  REST API to upload the files and make querys

(https://www.youtube.com/watch?v=p35GygHpxoI) in this video, the author share a repository with an streamlit frontend thats capable of receive files and then digest them to use

3 - Gradio UI 

Finally from this video (https://www.youtube.com/watch?v=QaDFz_vL4aw), create an Gradio UI to replace the streamlit mentioned before, with the capabilities to:

- handle upload files and index these files in weviate
- handle the query prompts

Im not shure if the control of env variables are so relevant, but for sure someone with ML knowledge can sugest some other free models to do the LLM job

So, thats it, hope this idea can help someone, and if you build or know something like this, please comment in this post :)

Sorry for any english mistake and Regards from Brazil!

**Comment by pseudotensor** [comment]
FYI, I love privateGPT and localGPT projects, h2oGPT has full UI and uploading of files currently: https://github.com/h2oai/h2ogpt
